src/schema/schema_alter.c

```diff
@@ -63,7 +54,164 @@ err:	__wt_free(session, config);
 	if (ret == WT_NOTFOUND)
 		ret = ENOENT;
 
-	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
+	return (ret);
+}
+
+/*
+ * __alter_file --
+ *	Alter a file.
+ */
+static int
+__alter_file(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_DECL_RET;
+	const char *filename, *uri;
+
+	/*
+	 * We know that we have exclusive access to the file.  So it will be
+	 * closed after we're done with it and the next open will see the
+	 * updated metadata.
+	 */
+	filename = uri = session->dhandle->name;
+	if (!WT_PREFIX_SKIP(filename, "file:"))
+		return (__wt_unexpected_object_type(session, uri, "file:"));
+
+	WT_RET(__wt_meta_track_on(session));
 
+	WT_ERR(__alter_apply(session,
+	    uri, cfg, WT_CONFIG_BASE(session, file_meta)));
+
+err:	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
 	return (ret);
 }
+
+/*
+ * __alter_tree --
+ *	Alter an index or colgroup reference.
+ */
+static int
+__alter_tree(WT_SESSION_IMPL *session, const char *name, const char *cfg[])
+{
+	WT_CONFIG_ITEM cval;
+	WT_DECL_ITEM(data_source);
+	WT_DECL_RET;
+	char *value;
+	bool is_colgroup;
+
+	value = NULL;
+
+	is_colgroup = WT_PREFIX_MATCH(name, "colgroup:");
+	if (!is_colgroup && !WT_PREFIX_MATCH(name, "index:"))
+		WT_ERR_MSG(session, EINVAL,
+		    "expected a 'colgroup:' or 'index:' source: '%s'", name);
+
+	/* Read the schema value. */
+	WT_ERR(__wt_metadata_search(session, name, &value));
+
+	/* Get the data source URI. */
+	if ((ret = __wt_config_getones(session, value, "source", &cval)) != 0)
+		WT_ERR_MSG(session, EINVAL,
+		    "index or column group has no data source: %s", value);
+
+	WT_ERR(__wt_scr_alloc(session, 0, &data_source));
+	WT_ERR(__wt_buf_fmt(session,
+	    data_source, "%.*s", (int)cval.len, cval.str));
+
+	/* Alter the data source */
+	WT_ERR(__wt_schema_alter(session, data_source->data, cfg));
+
+	/* Alter the index or colgroup */
+	if (is_colgroup)
+		WT_ERR(__alter_apply(session,
+		    name, cfg, WT_CONFIG_BASE(session, colgroup_meta)));
+	else
+		WT_ERR(__alter_apply(session,
+		    name, cfg, WT_CONFIG_BASE(session, index_meta)));
+
+err:	__wt_scr_free(session, &data_source);
+	__wt_free(session, value);
+	return (ret);
+}
+
+/*
+ * __alter_table --
+ *	WT_SESSION::alter for a table.
```

Comment:
This style of comment with WT_XXX::YYY is generally used for functions that are used as function pointers in handles - i.e it would be more appropriate as the comment for `__wt_schema_alter` than this function.

Reply:
fixed.